src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -43,6 +43,22 @@ internal class ManifestDocument
 			{ "action",   "android.intent.action.MAIN" },
 			{ "category", "android.intent.category.LAUNCHER" },
 		};
+
+		// the elements which we apply the "." -> PackageName replacement on
+		static readonly string [] ManifestElementFixupNames = {
```

Comment:
The "repeatedly update" means that we're actually *confirming* that the properties are correct. There are a number of elements which contain an `android:name` attribute which *should not* be updated, e.g. [`<category/>`](https://developer.android.com/guide/topics/manifest/category-element.html#nm).

I am thus inclined to prefer the `Dictionary<string, string[]>` approach.